pkg/querier/queryrange/util.go

```diff
@@ -70,3 +70,7 @@ func DoRequests(ctx context.Context, downstream Handler, reqs []Request, limits
 
 	return resps, firstErr
 }
+
+type Buffer interface {
```

Comment:
Why is Buffer such a long way away from where it gets used?
Do you even need a type?  `interface { Bytes() []byte }` is pretty straightforward.

Reply:
I can move it but I think it's nice to have the type so, Loki uses this package and can be used as documentation.